# **Quiz Instructions**
## **About this Quiz**

This is a closed-book, single-person quiz. Do not use any outside help or fill out questions with a partner. I will likely know. The quiz is worth 14 points. The quiz is three questions of varying point values. The quiz should not take more than 15 minutes for those without accommodations. Those with a 15 minute time limit, take care to start the quiz at least before 11:44 PM PT on quiz day, or you will not be afforded full time to finish. For those with time accommodations, be sure to before 11:36 PM PT on quiz day.

**About the Questions**

- Be sure to read the instructions for all questions carefully—I've taken great care to explain to you exactly what responses that will earn full-points should contain.



# **Question 1**

You are presented the following code. You are certain that cars is a Pandas DataFrame and is assigned car information. The two variables used, mpg and converters are also properly assigned and do not contain any problematic missing data.

Describe the expected output from the code below. Be as specific as you can about what the code might do. The best answers will go line-by-line and describe what each line of code does.

In [ ]:
import altair as alt
(
    alt.Chart(cars)
       .mark_point()
       .encode(
          x='mpg',
          y='converters'
       )
)

# **Answer 1**

```python
import altair as alt  # imports the altair module name as "alt"

(
    alt.Chart(cars) # using altair, create a chart object from the cars DataFrame
       .mark_point() # using that data, create a scatter plot
       .encode( # for the axis of the data
          x='mpg', # plot 'mpg' on the x axis
          y='converters' # plot 'converters' on the y axis
       )
)
```

# **Question 2**

Continuing from above, you receive the following lines of code from a colleague. Go line by line and explain what is happening in this code. Specifically, what do you expect the output of new_df to look like?

In [ ]:
cols = ['disp', 'mpg', 'converters']
new_df = cars[cols]

# **Answer 2**

In [ ]:
cols = ['disp', 'mpg', 'converters'] # from this list of columns
new_df = cars[cols] # a new dataframe of those columns will be formed. It is assigned to new_df and will only contain cols.

# **Question 3**

You receive the following Box and Whisker Plot of IMDb Scores from a colleague.

Identify the following features of the chart:

1. The median IMDb Score (1pt)
2. The 1st Quartile (1 pt)
3. The 3rd Quartile (1 pt)
4. Calculate the interquartile range (IQR) (3pts)

In [1]:
from os import environ
from google.colab import drive

from pandas import read_csv

# if we are starting here we need to remount our drive
drive.mount('/content/drive', force_remount=True)

file_path = "/content/drive/MyDrive/Work/Chapman/MGSC_310/MGSC_310_shared_files_and_resources/Data/IMDB_movies.csv"

movies = read_csv(file_path)

movies.head()

Mounted at /content/drive


movie_title      director_name      gross  \
0                                    Avatar      James Cameron  760505847   
1  Pirates of the Caribbean: At World's End     Gore Verbinski  309404152   
2                                   Spectre         Sam Mendes  200074175   
3                     The Dark Knight Rises  Christopher Nolan  448130642   
4                               John Carter     Andrew Stanton   73058679   

      budget country  title_year  imdb_score language  duration  \
0  237000000     USA        2009         7.9  English     178.0   
1  300000000     USA        2007         7.1  English     169.0   
2  245000000      UK        2015         6.8  English     148.0   
3  250000000     USA        2012         8.5  English     164.0   
4  263700000     USA        2012         6.6  English     132.0   

                            genres  ... director_facebook_likes  \
0  Action|Adventure|Fantasy|Sci-Fi  ...                       0   
1         Action|Adventure|Fantasy  ...                     563   
2        Action|Adventure|Thriller  ...                       0   
3                  Action|Thriller  ...                   22000   
4          Action|Adventure|Sci-Fi  ...                     475   

   cast_total_facebook_likes facenumber_in_poster actor_1_facebook_likes  \
0                       4834                  0.0                 1000.0   
1                      48350                  0.0                40000.0   
2                      11700                  1.0                11000.0   
3                     106759                  0.0                27000.0   
4                       1873                  1.0                  640.0   

      actor_1_name  actor_2_facebook_likes      actor_2_name  \
0      CCH Pounder                   936.0  Joel David Moore   
1      Johnny Depp                  5000.0     Orlando Bloom   
2  Christoph Waltz                   393.0      Rory Kinnear   
3        Tom Hardy                 23000.0    Christian Bale   
4     Daryl Sabara                   632.0   Samantha Morton   

   num_user_for_reviews  num_critic_for_reviews num_voted_users  
0                  3054                   723.0          886204  
1                  1238                   302.0          471220  
2                   994                   602.0          275868  
3                  2701                   813.0         1144337  
4                   738                   462.0          212204  

[5 rows x 25 columns]

In [8]:
import altair as alt
(
    alt.Chart(movies)
       .mark_boxplot()
       .encode(
          x='imdb_score'
       )
       .properties(
          width =400,
          height=200
       )
)

Output hidden; open in https://colab.research.google.com to view.

# **Answer 3**

1. Median = 6.6. anwhere above 6.5 (*not* inclusive) and below 6.7 (*not* inclusive) is fair game.

2. Q1 = 5.9. Anywhere above 5.75 (*not* inclusive) and below 6.0 (*not* inclusive) is fair game.

3. Q3 = 7.2. Anywhere below 7.35 (*not* inclusive) and above 7.0 (*not* inclusive) is fair game.

4. IQR = Q3 - Q1 = 1.3. But, using the bounds allowed from above, that means the upper bound for the answer is: 7.35 - 5.75 = 1.6. The lower bound for the answer would be 7 - 6 = 1.0.

  Altogether, your answer must be: $$1.0 < IQR < 1.6$$